packages/flutter/test/painting/border_test.dart

```diff
@@ -299,5 +298,108 @@ void main() {
     const BorderSide outsideSide = BorderSide(width: 10, strokeAlign: BorderSide.strokeAlignOutside);
     const BorderDirectional outsideBorderDirectional = BorderDirectional(top: outsideSide, bottom: outsideSide, start: outsideSide, end: outsideSide);
     expect(outsideBorderDirectional.dimensions, EdgeInsetsDirectional.zero);
+
+    const Border nonUniformBorder = Border(
+      left: BorderSide(width: 5),
+      top: BorderSide(width: 10, strokeAlign: BorderSide.strokeAlignCenter),
+      right: BorderSide(width: 15, strokeAlign: BorderSide.strokeAlignOutside),
+      bottom: BorderSide(width: 20),
+    );
+    expect(nonUniformBorder.dimensions, const EdgeInsets.fromLTRB(5, 5, 0, 20));
+
+    const BorderDirectional nonUniformBorderDirectional = BorderDirectional(
+      start: BorderSide(width: 5),
+      top: BorderSide(width: 10, strokeAlign: BorderSide.strokeAlignCenter),
+      end: BorderSide(width: 15, strokeAlign: BorderSide.strokeAlignOutside),
+      bottom: BorderSide(width: 20),
+    );
+    expect(nonUniformBorderDirectional.dimensions, const EdgeInsetsDirectional.fromSTEB(5, 5, 0, 20));
+  });
+
+  testWidgets('Non-Uniform Border variations', (WidgetTester tester) async {
+
+    Widget buildWidget({ required BoxBorder border, BorderRadius? borderRadius, BoxShape boxShape = BoxShape.rectangle}) {
+      return Directionality(
+        textDirection: TextDirection.ltr,
+        child: DecoratedBox(
+          decoration: BoxDecoration(
+            shape: boxShape,
+            border: border,
+            borderRadius: borderRadius,
+          ),
+        ),
+      );
+    }
+
+    // This is used to test every allowed non-uniform border combination.
+    const Border allowedBorderVariations = Border(
+      left: BorderSide(width: 5),
+      top: BorderSide(width: 10, strokeAlign: BorderSide.strokeAlignCenter),
+      right: BorderSide(width: 15, strokeAlign: BorderSide.strokeAlignOutside),
+      bottom: BorderSide(width: 20),
+    );
+
+    // This falls into non-uniform border because of strokeAlign.
+    await tester.pumpWidget(buildWidget(border: allowedBorderVariations));
+    expect(tester.takeException(), isNull,
+        reason: 'Border with non-uniform strokeAlign should not fail.');
+
+    await tester.pumpWidget(buildWidget(
+      border: allowedBorderVariations,
+      borderRadius: BorderRadius.circular(25),
+    ));
+    expect(tester.takeException(), isNull);
+
+    await tester.pumpWidget(buildWidget(border: allowedBorderVariations, boxShape: BoxShape.circle));
+    expect(tester.takeException(), isNull);
+
+    await tester.pumpWidget(
+      buildWidget(
+        border: const Border(
+          left: BorderSide(width: 5, style: BorderStyle.none),
+          top: BorderSide(width: 10),
+          right: BorderSide(width: 15),
+          bottom: BorderSide(width: 20),
+        ),
+        borderRadius: BorderRadius.circular(25),
+      ),
+    );
+    expect(tester.takeException(), isNotNull,
```

Comment:
I'd use `isAssertionError` instead of `isNotNull` to be more specific. Here and below.

Reply:
Thanks. I didn't know about this.